### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by knopki**
Almost total reimplementation. `yascheduler` now is total asynchronous via python's `asyncio`. `paramiko` and `plumbum` is dropped. `asyncssh` is used instead. Also, no more thread pools of background workers.

Configuration is parsed, validated and typed in one place. Config is almost the same. Some additional parameters added - see `README.md`. Also, some not backward compatible changes for Azure provider.

Database access is abstracted in one place - `yascheduler/db.py`. All queries are copied almost as is. Results are mapped to simple models for node and task. API is asynchronous. Internal synchronous code runs in one-thread-excutor because `pg8000` is not thread-safe. Small schema change - `username` column in `yascheduler_nodes` table. Automatic migration.

Rich `RemoteMachine` object is used instead of just ssh connection. `RemoteMachine` is self configured abstraction over remote ssh-connected machine. Supports basic operations like run command, run command in background, SFTP operations, pgrep. Windows is supported! All operations are asynchronous. Yes, many operations in parallel via different channels of the same connections. Connect via jump box is supported too. Support background checks of task running status.

`daemonize()` moved to `utils.py` and installed as `yascheduler` on package installation.

Hetzner and Upcload clouds support implementations almost not changed, but asynchronized, so multiple allocation can be in flight at the same time. Also, server type and image can be configured for Hetzner.

Azure support implementation is different. Instead of trying to provision all cloud configuration, cloud pre-configuration is implied. Instructions in `CLOUD.md`. Deployment (in Azure meaning) are not used anymore. Just network interface and VM created and deleted later. IPs are internal.

When node scheduled for allocation in a cloud, cloud is elected. Clouds without free capacity are filtered out. Clouds with unsupported platforms are filtered out too. Other clouds are sorted by priority.

Scheduler daemon is built with idea of multiple producers-consumers:
- producer of IPs from db to connect to and multiple consumers that connect to remote machines
- producer of tasks to scheduler and multiple consumers that tries to allocate task to free remote machine or allocate new node in the cloud
- producer of completed tasks to download and multiple consumers that downloads task if it is ready
- producer of idle nodes to de-allocate and multiple consumers that de-allocates nodes

New tasks are allocated to recently freed machines. The first to be deleted are the machines that have been free for a longest time. In this way, the minimum required number of machines is maintained at all times.

### CODE REVIEW

**Comment by knopki** [comment]
Still not ready. Need to finish Azure adapter.

**Comment by knopki** [comment]
Things to do:
- port #55
- create simple client synchronous client library for scheduler

**Comment by knopki** [comment]
`Yascheduler` renamed to `Scheduler`, because it's not a public class, it's an internal daemon.
`Yascheduler` class introduced with synchronous interface to enqueue and query tasks. Signature is almost the same. See examples.
@blokhin plz review

File: yascheduler/remote_machine/remote_machine.py [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+    Optional,
+    Pattern,
+    Sequence,
+    Set,
+    Type,
+    Union,
+)
+
+import asyncssh
+import asyncstdlib as a
```
Comment (blokhin) [inline yascheduler/remote_machine/remote_machine.py:22]: @knopki NB one-letter variable `a`

**Review by blokhin — APPROVED** [review]

**Pull request merged**